testdata/p4_16_samples/psa-resubmit-bmv2.p4

```diff
@@ -0,0 +1,172 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <core.p4>
+#include "psa.p4"
+
+
+typedef bit<48>  EthernetAddress;
+
+header ethernet_t {
+    EthernetAddress dstAddr;
+    EthernetAddress srcAddr;
+    bit<16>         etherType;
+}
+
+struct empty_metadata_t {
+}
+
+struct metadata_t {
+}
+
+struct headers_t {
+    ethernet_t       ethernet;
+}
+
+parser IngressParserImpl(packet_in pkt,
+                         out headers_t hdr,
+                         inout metadata_t user_meta,
+                         in psa_ingress_parser_input_metadata_t istd,
+                         in empty_metadata_t resubmit_meta,
+                         in empty_metadata_t recirculate_meta)
+{
+    state start {
+        pkt.extract(hdr.ethernet);
+        transition accept;
+    }
+}
+
+control cIngress(inout headers_t hdr,
+                 inout metadata_t user_meta,
+                 in    psa_ingress_input_metadata_t  istd,
+                 inout psa_ingress_output_metadata_t ostd)
+{
+
+    action inc() {
+      hdr.ethernet.dstAddr = hdr.ethernet.dstAddr + hdr.ethernet.srcAddr;
+    }
+    action resubmit() {
+      ostd.resubmit = true;
+    }
+    action stop() {
+        ostd.resubmit = false;
+    }
+    table t {
+      actions = { inc;
+                }
+      default_action = inc;
+    }
+
+    table debug {
+        key = {
+            hdr.ethernet.dstAddr: exact;
+        }
+        actions = { NoAction; }
+        default_action = NoAction;
+    }
+
+    table t2 {
+        actions = { resubmit; }
+        default_action = resubmit;
+    }
+
+    table t3 {
+        actions = { stop; }
+        default_action = stop;
+    }
+
+    apply {
+        t.apply();
+        debug.apply();
+        if (hdr.ethernet.dstAddr >= 48w4) {
+            t3.apply();
+            send_to_port(ostd, (PortId_t) ((PortIdUint_t) hdr.ethernet.dstAddr[3:0]));
+            if (hdr.ethernet.dstAddr >= 48w4) {
+                hdr.ethernet.dstAddr = 48w5;
+            }
+        }
+        else{
+          t2.apply();
+        }
+    }
+}
+
+parser EgressParserImpl(packet_in buffer,
+                        out headers_t hdr,
+                        inout metadata_t user_meta,
+                        in psa_egress_parser_input_metadata_t istd,
+                        in empty_metadata_t normal_meta,
+                        in empty_metadata_t clone_i2e_meta,
+                        in empty_metadata_t clone_e2e_meta)
+{
+    state start {
```

Comment:
right before `transition accept;`, we need to add a `buffer.extract(hdr.ethernet);`